### PR TITLE
store: fix snap status set order.

### PR DESCRIPTION
**Pull request by siddontang**
If schedule quickly run, the snap status will be set SnapState::Snap first, and then changed to SnapState::Generating, so we can't do any snapshot later. 

We must set SnapState::Generating before schedule.

Notice: this can't fix whole ci.

@ngaut @BusyJay

### CODE REVIEW

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -222,9 +222,10 @@ impl<T: Transport, C: PdClient> Store<T, C> {
                 debug!("handling snapshot for {}", region_id);
                 let task = SnapTask::new(peer.storage.clone());
                 debug!("task generated");
-                match self.snap_worker.schedule(task) {
-                    Err(e) => error!("failed to schedule snap task {}", e),
-                    Ok(_) => peer.storage.wl().snap_state = SnapState::Generating,
+                peer.storage.wl().snap_state = SnapState::Generating;
```
Comment (tiancaiamao) [inline src/raftstore/store/store.rs]: this change would overwrite the previous snap_state, will it cause trouble somewhere else?
...is the snap processing serialization and the previous result has been processed(notified)

**Comment by ngaut** [comment]
LGTM

File: src/raftstore/store/store.rs (hunk 1 above)
Comment (ngaut) [inline src/raftstore/store/store.rs]: Do we have that problem ? @siddontang

Comment (BusyJay) [inline src/raftstore/store/store.rs]: This is processing the previous state (by setting a new state).

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**